src/Aspire.Hosting/MongoDB/MongoDBBuilderExtensions.cs

```diff
@@ -66,18 +68,20 @@ public static IResourceBuilder<MongoDBDatabaseResource> AddDatabase(this IResour
         var mongoExpressContainer = new MongoExpressContainerResource(containerName);
         builder.ApplicationBuilder.AddResource(mongoExpressContainer)
                                   .WithAnnotation(new ContainerImageAnnotation { Image = "mongo-express", Tag = "1.0.2-20" })
-                                  .WithEnvironment(context => ConfigureMongoExpressContainer(context, builder.Resource))
+                                  .WithEnvironment(context => ConfigureMongoExpressContainer(builder.ApplicationBuilder.Configuration, context, builder.Resource))
                                   .WithHttpEndpoint(containerPort: 8081, hostPort: hostPort, name: containerName)
                                   .ExcludeFromManifest();
 
         return builder;
     }
 
-    private static void ConfigureMongoExpressContainer(EnvironmentCallbackContext context, IResource resource)
+    private static void ConfigureMongoExpressContainer(IConfiguration configuration, EnvironmentCallbackContext context, IResource resource)
     {
+        var containerHostName = HostNameResolver.ReplaceLocalhostWithContainerHost("localhost", configuration);
```

Comment:
Might be a dumb question but don't we also need to update the schema for this change?

Reply:
What schema?

Reply:
nvm I think I was misunderstanding. I thought this was adding an option into the component's json configuration shchema, but doesn't look like that's the case.